.github/workflows/ci_cd_test.yml

```diff
@@ -5,108 +5,99 @@ name: PR build & test
+    paths:
+      - "lyncs*/__init__.py"
 
 jobs:
-  build-n-publish:  
-    runs-on: ubuntu-latest
-
+  build-n-publish:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        py-version:
+          - 3.x
+        mpi:
+          - openmpi
+        os:
+          - ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-      with:
-        ref: ${{ github.head_ref }}
-
-    - name: Set up Python
-      uses: actions/setup-python@v1
-      with:
-        python-version: '3.x'
-        
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        
-    - name: Applying black formatting
-      run: |
-        pip install black
-        black --diff .
-        black .
-
-    - name: Pushing changes if any
-      uses: stefanzweifel/git-auto-commit-action@v4
-      with:
-        commit_message: Applying black formatting (from Github Action)
-        commit_user_name: sbacchio
-        commit_user_email: s.bacchio@gmail.com
-        commit_author: Simone Bacchio <s.bacchio@gmail.com>
-
-    - name: Install from source
-      run: |
-        pip install -e .[all]
-
-    - name: Pylint output
-      run: |
-        pip install lyncs_setuptools[pylint]
-        badge=$(lyncs_pylint_badge --fail-under 8 . | sed "s/\&/\\\&/g")
-        badge_line=$(awk '/!\[pylint\]/ {print FNR}' README.md)
-        sed -i "${badge_line}s#.*#${badge}#" README.md
-
-    - name: Pushing changes if any
-      uses: stefanzweifel/git-auto-commit-action@v4
-      with:
-        commit_message: Updating pylint score (from Github Action)
-        commit_user_name: sbacchio
-        commit_user_email: s.bacchio@gmail.com
-        commit_author: Simone Bacchio <s.bacchio@gmail.com>
-
-    - name: Run tests
-      run: |
-        pytest -v
-
-    - name: Run lyncs_setuptools
-      run: |
-        lyncs_setuptools
-
-    - name: Upload if not up to date
-      env:
-        TWINE_USERNAME: __token__
-        TWINE_PASSWORD: ${{ secrets.test_pypi_password }}
-      run: |
-        pip uninstall -y lyncs_io
-        pip install --extra-index-url https://test.pypi.org/simple/ lyncs_io==$(lyncs_setuptools version) || (
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Install MPI (${{ matrix.mpi }})
+        run: bash .bash/install-${{ matrix.mpi }}.sh
+
+      - name: Set up Python ${{ matrix.py-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.py-version }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install lyncs_setuptools
+
+      - name: Upload if not up to date
+        id: check-version
+        continue-on-error: true
+        run: 
+          pip install --extra-index-url https://test.pypi.org/simple/ lyncs_io==$(lyncs_setuptools version)
+
+      - name: Install from source
+        if: ${{ steps.check-version.conclusion == "failure" }}
```

Comment:
I don't understand why this is a problem.
Should work like: https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#steps-context
But at the moment I get a syntax error: https://github.com/Lyncs-API/lyncs.io/actions/runs/679019981/workflow

Reply:
Not sure if it makes a difference but you could try single quotations i.e 'failure'?